### ISSUE TITLE
Refresh credentials upon each attempt to access sources 

### ISSUE THREAD

**Issue by medriscoll** [rilldata/rill#3478]
**Describe the bug**
At present, if an update is made to one's local AWS credentials, Rill must be restarted in order for those updated credentials to be recognized.

**To Reproduce**
1. Attempt to add a source in Rill pointing at a private S3 bucket that requires credentials
2. Receive an access error
3. Update the credentials in ~/.aws/credentials
4. Re-attempt to add a source pointing to the same S3 bucket
5. Receive the same access error

**Expected behavior**
The access should work on the second attempt, following a change to the local credentials in ~/.aws/credentials.

**Comment by k-anshul** [comment]
Hey @medriscoll 

Could you please share more details on how you are updating credentials? It is working for me. I tried two approaches:
1. Updating `~/.aws/credentials` via `aws configure`.
2. Updating `~/.aws/credentials` directly via file rename.

**Comment by nishantmonu51** [comment]
@k-anshul : Have you tried reproducing it without restarting Rill ?

**Comment by k-anshul** [comment]
> @k-anshul : Have you tried reproducing it without restarting Rill ?

Yes. I tried the exact same steps that Mike mentioned. Tried two ways for step 3 (`Update the credentials in ~/.aws/credentials`) i.e. via aws configure and direct file edit and both worked.
